shared/store/userPreferences.ts

```diff
@@ -0,0 +1,55 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { LocalStorageProvider } from '@api/local_storage/provider'
+import { EnvironmentNetwork } from '@environment'
+
+export interface LabeledAddress {
+  [address: string]: string
+}
+
+export interface UserPreferences {
+  addresses: LabeledAddress
+}
+
+const initialState: UserPreferences = {
+  addresses: {}
+}
+
+export const fetchUserPreferences = createAsyncThunk(
+  'userPreferences/fetchUserPreferences',
+  async (network: EnvironmentNetwork) => {
+    return await LocalStorageProvider.getUserPreferences(network)
+  }
+)
+
+export const setUserPreferences = createAsyncThunk(
+  'userPreferences/setUserPreferences',
+  async ({
+    network,
+    preferences
+  }: { network: EnvironmentNetwork, preferences: UserPreferences }) => {
+    await LocalStorageProvider.setUserPreferences(network, preferences)
+  }
+)
+
+export const setAddresses = createAsyncThunk(
+  'userPreferences/setAddresses',
+  async (addresses: LabeledAddress) => {
+    return addresses
```

Comment:
Since we are not performing any async task here, think we don't need `createAsyncThunk`. Can we move the logic from `extraReducers` to `reducer`?

Reply:
I moved it here because I wanted to make it an async thunk, meaning, after you set the address via dispatch, you will then call the setUserPrefences (to store it in the local storage). I'm not sure if we move it to extra reducer, it will work. But we can test it out once there's a UI side to integrate (set address flow).